datalad/interface/common_opts.py

```diff
@@ -324,6 +329,7 @@
         constraints=EnsureChoice(*list(known_result_xfms.keys())) | EnsureCallable() | EnsureNone()),
     result_renderer=Parameter(
         doc="""format of return value rendering on stdout""",
+        default='default',
```

Comment:
As far as I can tell, this particular line is unnecessary (see script below). Adding it seems likely to confuse readers given that these `common_opts` parameters apply to both the Python and command-line interface, but the default `result_renderer` value for Python callers is `None`.

<details>
<summary>script</summary>

```sh
set -eu

base=0c6d355af98f0e1dfccd96c84cc77c5ab7a9121d
pr=02fa5192d99e0c49e9c23c87d1546fd82015066b

doc_dump () {
    commit="$1"
    git checkout --quiet "$commit"
    python -m pydoc datalad.api.push >pydoc-output-"$commit" 2>&1
    datalad --help >help-output-"$commit" 2>&1
}

doc_diff () {
    reva="$1"
    revb="$2"
    git diff --no-index pydoc-output-"$reva" pydoc-output-"$revb" || :
    git diff --no-index help-output-"$reva" help-output-"$revb" || :

}

doc_dump "$base"
doc_dump "$pr"
doc_diff "$base" "$pr"

pr_revised=9ae17ce4fc44447b005ce87fcc53f1280aef1572
git diff "$pr" "$pr_revised"

echo "No changes:"
doc_dump "$pr_revised"
doc_diff "$pr" "$pr_revised"
```

```
diff --git a/help-output-0c6d355af98f0e1dfccd96c84cc77c5ab7a9121d b/help-output-02fa5192d99e0c49e9c23c87d1546fd82015066b
index c226aa258e..dc13507ec7 100644
--- a/help-output-0c6d355af98f0e1dfccd96c84cc77c5ab7a9121d
+++ b/help-output-02fa5192d99e0c49e9c23c87d1546fd82015066b
@@ -160,7 +160,7 @@ specific --help, i.e.: datalad <command> --help
                         '{metadata[name]}'. If a 2nd-level key contains a
                         colon, e.g. 'music:Genre', ':' must be substituted by
                         '#' in the template, like so:
-                        '{metadata[music#Genre]}'.
+                        '{metadata[music#Genre]}'. [Default: 'default']
   --report-status {success,failure,ok,notneeded,impossible,error}
                         constrain command result report to records matching
                         the given status. 'success' is a synonym for 'ok' OR
@@ -177,7 +177,8 @@ specific --help, i.e.: datalad <command> --help
                         'continue' works like 'ignore', but an error causes a
                         non-zero exit code; 'stop' halts on first failure and
                         yields non-zero exit code. A failure is any result
-                        with status 'impossible' or 'error'.
+                        with status 'impossible' or 'error'. [Default:
+                        'continue']
   --cmd                 syntactical helper that can be used to end the list of
                         global command line options before the subcommand
                         label. Options taking an arbitrary number of arguments
diff --git a/datalad/interface/common_opts.py b/datalad/interface/common_opts.py
index 6246085674..fdba2e90e3 100644
--- a/datalad/interface/common_opts.py
+++ b/datalad/interface/common_opts.py
@@ -329,7 +329,6 @@
         constraints=EnsureChoice(*list(known_result_xfms.keys())) | EnsureCallable() | EnsureNone()),
     result_renderer=Parameter(
         doc="""format of return value rendering on stdout""",
-        default='default',
         constraints=EnsureChoice('default', 'json', 'json_pp', 'tailored') | EnsureNone()),
     on_failure=Parameter(
         doc="""behavior to perform on failure: 'ignore' any failure is reported,
No changes:
```

</details>